### PR TITLE
Testcases to reboot/shutdown and recovery node when amq workload is running

### PR DESCRIPTION
**Pull request by Akarsha-rai**
Signed-off-by: Akarsha <akrai@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/amq.py [hunk 1]
```diff
@@ -759,3 +760,36 @@ def cleanup(
...
+
+    Returns:
+        list: Cluster node OCP objects
+
+    """
+    # Get the worker node list
+    worker_nodes = get_typed_nodes(node_type='worker')
+
+    # Get the node name
+    worker_node = [node.get().get('metadata').get('name') for node in worker_nodes]
```
Comment (ebenahar) [inline ocs_ci/ocs/amq.py:783]: the var should reflect that the value is a list.
As you are not using `worker_nodes` var in other places, I think you can do:
```python
worker_nodes = [node.get().get('metadata').get('name') for node in get_typed_nodes(node_type='worker')]
```

File: tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        result = self.thread.result(timeout=1800)
+        log.info(result)
+        assert self.amq.validate_amq_benchmark(
+            result=result, amq_workload_yaml=self.amq_workload_dict
+        ) is not None, (
+            "Benchmark did not completely run or might failed in between"
+        )
+
+    @pytest.mark.polarion_id("OCS-1282")
+    def test_amq_after_rebooting_worker_node(self, nodes, amq_setup):
```
Comment (PrasadDesala) [inline tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py:137]: The sequence of steps seems similar, can we use pytest.param here?
@pytest.mark.parametrize(
        argnames=["node_type"],
        argvalues=[
            pytest.param(*['worker'], marks=pytest.mark.polarion_id("xxx")),
            pytest.param(*['master'], marks=pytest.mark.polarion_id("xxx"))
        ]
    )

File: tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        amq workloads running in background
+
+        """
+        # Get all amq pods
+        pod_obj_list = get_all_pods(namespace=constants.AMQ_NAMESPACE)
+
+        # Get the master node list
+        master_nodes = get_typed_nodes(node_type='master')
+
+        # Reboot one master nodes
```
Comment (PrasadDesala) [inline tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py:95]: get_typed_nodes can take the num of nodes as input,  can selecting a node by random choice be replaced with that?

File: tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15)(
+            ocp.wait_for_cluster_connectivity(tries=400)
+        )
+        retry(
+            (CommandFailed, TimeoutError, AssertionError, ResourceWrongStatusException),
+            tries=60,
+            delay=15)(
+            wait_for_nodes_status(timeout=1800)
```
Comment (PrasadDesala) [inline tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py:115]: the timeout is quite high, any reason for such a very long value?

File: tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py (hunk 2 above)
Comment (Akarsha-rai) [reply tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py]: done

File: tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py (hunk 3 above)
Comment (Akarsha-rai) [reply tests/e2e/workloads/amq/test_amq_node_reboot_and_shutdown.py:100]: done

**Comment by Akarsha-rai** [comment]
When one of the nodes is rebooted, amq benchmark stops working and it doesn't survive failures so as discussed replaced AMQ benchmark as workload generator in disruptive testing with Producer & Consumer APIs. Updated these changes in existing testcase and fixture, locally verified.

**Pull request merged**